plugin/src/main/java/org/opensearch/ml/action/profile/MLTaskProfileAction.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.action.profile;
+
+import org.opensearch.action.ActionType;
+import org.opensearch.ml.constant.CommonValue;
+
+public class MLTaskProfileAction extends ActionType<MLTaskProfileResponse> {
+    // Internal Action which is not used for public facing RestAPIs.
+    public static final String NAME = CommonValue.ACTION_PREFIX + "tasks/profile";
```

Comment:
How about "tasks/profile" -> "profile" as we are going to profile not just for task, also for model and may add other things in future.